### ISSUE TITLE
Add tuple extensions for Result type

### ISSUE THREAD

**Issue by bartsokol** [MonacsLib/Monacs#1]
Add overloads at least for Bind and Map extension methods for Result type that accept `Result` of tuple (`Result<System.ValueTuple>` and `Result<System.Tuple>`) and deconstruct them as binder/mapper parameters. For now, 2- and 3-element tuples should be covered. I would suggest using names like `Bind2` and `Bind3` to distinguish them from regular variants (just like F# does it, but it's open for discussion). Add them in `Monacs.Core.Tuples` namespace, in a same way as `Unit` and async extensions are added.

One example could be:
```
public static Result<TResult> Bind2<TResult, T1, T2>(this Result<(T1, T2)> result, Func<T1, T2, Result<TResult>> func) =>
            result.IsOk ? func(result.Value.Item1, result.Value.Item2) : Error<TResult>(result.Error);
```

All the functions should be documented using XML docs in the same way as the other extensions will be in version 0.2 (WIP). Unit tests need to be added for the extensions as well.

Use [tuple-extensions branch](https://github.com/MonacsLib/Monacs/tree/tuple-extensions) as a staring point.

**Comment by pizycki** [comment]
dibs on !

**Comment by pizycki** [comment]
Here is what I have already
 * `Result.Match2` and `3`
 * `Result.MatchTo2`
 * `Result.Do2` and `3`
 * `Result.TryCatch2` and `3`

All extensions are covered with tests.

TBD:
- [ ] More extensions
- [ ] Change extensions location and namespaces
- [ ] Add in-code doc

**Comment by bartsokol** [comment]
PR is merged but Travis CI build is still failing (due to issues with `System.ValueTuple`), looking to find a solution for it.

**Comment by bartsokol** [comment]
Build is fixed, tests are added, will be released in version 0.3.
